module/VuFind/src/VuFind/Recommend/LibGuidesProfile.php

```diff
@@ -0,0 +1,241 @@
+     * Map of LibGuides subject name (lowercase) to ID of an account
+     * (presumably a librarian) who wrote its subject guide
+     *
+     * @var array
+     */
+    protected $subjectToId = [];
+
+    /**
+     * LibGuides connector
+     *
+     * @var LibGuides
+     */
+    protected $libGuides;
+
+    /**
+     * Constructor
+     *
+     * @param array      $config Config object representing LibGuidesAPI.ini
+     * @param HttpClient $client VuFind HTTP client
+     */
+    public function __construct($config, HttpClient $client)
+    {
+        $this->libGuides = new LibGuides(
+            $client,
+            $config->General->api_base_url,
+            $config->General->client_id,
+            $config->General->client_secret
+        );
+    }
+
+    /**
+     * Store the configuration of the recommendation module.
+     *
+     * @param string $settings Settings from searches.ini.
+     *
+     * @return void
+     */
+    public function setConfig($settings)
+    {
+        // No action needed.
+    }
+
+    /**
+     * Called before the Search Results object performs its main search
+     * (specifically, in response to \VuFind\Search\SearchRunner::EVENT_CONFIGURED).
+     * This method is responsible for setting search parameters needed by the
+     * recommendation module and for reading any existing search parameters that may
+     * be needed.
+     *
+     * @param \VuFind\Search\Base\Params $params  Search parameter object
+     * @param \Laminas\Stdlib\Parameters $request Parameter object representing user
+     * request.
+     *
+     * @return void
+     *
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function init($params, $request)
+    {
+        // No action needed.
+    }
+
+    /**
+     * Called after the Search Results object has performed its main search.  This
+     * may be used to extract necessary information from the Search Results object
+     * or to perform completely unrelated processing.
+     *
+     * @param \VuFind\Search\Base\Results $results Search results object
+     *
+     * @return void
+     */
+    public function process($results)
+    {
+        $this->results = $results;
+    }
+
+    /**
+     * Get terms related to the query.
+     *
+     * @return array
+     */
+    public function getResults()
+    {
+        // TODO: cache data, check it for staleness
+        $this->refreshData();
+
+        $query = $this->results->getParams()->getQuery();
+        $account = $this->findBestMatch($query);
+        return $account;
+    }
+
+    /**
+     * Find the VuFind account whose profile best matches the
+     * given query.
+     *
+     * @param \VuFindSearch\Query\QueryInterface $query Current search query
+     *
+     * @return array LibGuides account
+     */
+    protected function findBestMatch(\VuFindSearch\Query\QueryInterface $query)
+    {
+        $queryString = $query->getAllTerms();
+        if (!$queryString) {
+            return false;
```

Comment:
Well, by "very generalized searches" I really mean "blank searches that bring back everything." :-)

Reply:
Ah sorry, I didn't look closely enough at the context :)  I can see this both ways.  On the one hand the levenshtein is already dumb enough to return a bad match rather than no match in difficult cases.  On the other hand there really is no good match for an empty search, and I feel it would make it look like the recommendation is meaningless or random.  It might be useful to recommend something generic like "call the help desk", but our template will already have plenty of generic links for that.  So I think I prefer to skip the recommendation here.

Reply:
That's fine, but it's certainly a possible future enhancement to have a special generic display if there's no specific recommendation found. Of course, this could easily be achieved with a custom template and the current code by adding an else clause, so maybe that's enough.